File: Java/src/com/dbc/algorithm/DynamicProgramming/LongestTurbulentSubarray.java
package com.dbc.algorithm.DynamicProgramming;

public class LongestTurbulentSubarray {
    public static int maxTurbulenceSize(int[] arr) {
        int ne = 1, op = 1, result = 1;
        for (int i = 1; i < arr.length; i++) {
            if (arr[i] < arr[i - 1]) {
                op = ne + 1;
                ne = 1;
            } else if (arr[i] > arr[i - 1]) {
                ne = op + 1;
                op = 1;
            } else {
                ne = 1;
                op = 1;
            }

            result = Math.max(result, Math.max(ne, op));
        }

        return result;
    }
}

File: Python/dynamic-programming/longest_turbulent_subarray.py
from typing import List


def maxTurbulenceSize(arr: List[int]) -> int:
    n = len(arr)
    op, ne, result = 1, 1, 1
    for i in range(1, n):
        if arr[i] > arr[i - 1]:
            ne = op + 1
            op = 1
        elif arr[i] < arr[i - 1]:
            op = ne + 1
            ne = 1
        elif arr[i] == arr[i - 1]:
            op = 1
            ne = 1
        
        result = max(result, ne, op)

    return result


if __name__ == "__main__":
    print(maxTurbulenceSize([100]))
